tools/flaky-test-reporter/jsonreport/jsonreport.go

```diff
@@ -41,13 +41,27 @@ type Report struct {
 	Flaky []string `json:"flaky"`
 }
 
+type JSONClient interface {
+	CreateReportForRepo(repo string, flaky []string, writeFile bool) (*Report, error)
+	ParseFlakyLog(repo string, buildID int, f func(report Report, result *[]string)) ([]Report, []string, error)
+	GetFlakyTestReport(repo string, buildID int) ([]Report, error)
+}
+
+// Client is simply a way to call methods, it does not contain any data itself
+type Client struct{}
```

Comment:
why do we have both JsonClient and Client? Are we expecting to add other clients? then, the methods should be on the client interface and JSONClient should be the implementation.